src/client/js/CoreEditor.js

```diff
@@ -587,37 +545,37 @@ function findLinkEntities(contentBlock, callback) {
 class Link extends React.Component {
     constructor(props) {
         super(props);
+        this.getProps(props);
+    }
+
+    getProps(props) {
         const {
-            linkTitle, linkUrl, linkDesc, onLinkEdit
-        } = Entity.get(props.entityKey).getData();
+            linkTitle, linkUrl, linkDesc, editCallback
+            } = Entity.get(props.entityKey).getData();
+        console.log(Entity.get(props.entityKey).getData());
         this.linkTitle = linkTitle;
         this.linkUrl = linkUrl;
         this.linkDesc = linkDesc;
-        this.onLinkEdit = (e) => {
-            e.preventDefault();
-            onLinkEdit(this.linkTitle, this.linkUrl, this.linkDesc, props.entityKey);
-        };
+        this.entityKey = props.entityKey;
+        this.linkCallback = editCallback;
     }
+
+    onLinkEdit (e) {
+        e.preventDefault();
+        this.linkCallback(this.linkTitle, this.linkUrl, this.linkDesc, this.entityKey);
+    }
+
     componentWillReceiveProps(props) {
-        const {
-            linkTitle, linkUrl, linkDesc, onLinkEdit
-        } = Entity.get(props.entityKey).getData();
-        this.linkTitle = linkTitle;
-        this.linkUrl = linkUrl;
-        this.linkDesc = linkDesc;
-        this.onLinkEdit = (e) => {
-            e.preventDefault();
-            onLinkEdit(this.linkTitle, this.linkUrl, this.linkDesc, props.entityKey);
-        };
```

Comment:
good refactoring practice - remove code doubles, we have same code in constructor and componentWillReceiveProps, let's move it in new method, makes sense?